test/e2e/dataplane/service_discovery.go

```diff
@@ -42,4 +53,78 @@ func RunServiceDiscoveryTest(f *framework.Framework) {
 
 	Expect(err).NotTo(HaveOccurred())
 	Expect(stdout).To(ContainSubstring("Welcome to nginx!"))
+
+	By(fmt.Sprintf("Executing dig to verify Nginx service is discoverable and service IP in cluster %q is returned", framework.ClusterC))
+	verifyClusterIpWithDig(f, framework.ClusterB, nginxServiceClusterC, netshootPodList, true)
+
+	By(fmt.Sprintf("Deleting Nginx service %q", nginxServiceClusterC.Name))
+	f.DeleteService(framework.ClusterC, nginxServiceClusterC.Name)
+
+	By(fmt.Sprintf("Executing dig to verify Nginx service is no longer discoverable"))
+	verifyClusterIpWithDig(f, framework.ClusterB, nginxServiceClusterC, netshootPodList, false)
+}
+
+func RunServiceDiscoveryLocalTest(f *framework.Framework) {
+	clusterBName := framework.TestContext.KubeContexts[framework.ClusterB]
+	clusterCName := framework.TestContext.KubeContexts[framework.ClusterC]
+
+	By(fmt.Sprintf("Creating an Nginx Deployment on on %q", clusterBName))
+	f.NewNginxDeployment(framework.ClusterB)
+
+	By(fmt.Sprintf("Creating a Nginx Service on %q", clusterBName))
+	nginxServiceClusterB := f.NewNginxService(framework.ClusterB)
+
+	By(fmt.Sprintf("Creating an Nginx Deployment on on %q", clusterCName))
+	f.NewNginxDeployment(framework.ClusterC)
+
+	By(fmt.Sprintf("Creating a Nginx Service on %q", clusterCName))
+	nginxServiceClusterC := f.NewNginxService(framework.ClusterC)
+
+	By(fmt.Sprintf("Creating a Netshoot Deployment on %q", clusterBName))
+	netshootPodList := f.NewNetShootDeployment(framework.ClusterB)
+
+	By(fmt.Sprintf("Executing dig to verify Nginx service is discoverable and service IP in cluster %q is returned", clusterBName))
+	verifyClusterIpWithDig(f, framework.ClusterB, nginxServiceClusterB, netshootPodList, true)
+
+	By(fmt.Sprintf("Deleting Nginx service %q", nginxServiceClusterB.Name))
```

Comment:
```suggestion
	By(fmt.Sprintf("Deleting Nginx service %q on %q", nginxServiceClusterB.Name, clusterBName))
```

Reply:
Looks like you missed this change. In fact we should just output the message in `DeleteService`:

`By(fmt.Sprintf("Deleting service %q on %q", nginxServiceClusterC.Name, TestContext.KubeContexts[cluster]))`

Reply:
Moved to DeleteService